site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+reason, when a config conflict is detected, implementations should identify the
+config elements that conflict, including the routes that are the original source
+of each element, and provide that info to users via appropriate methods of
+notification  (e.g., logs, message in GUI, etc.).
+
+NOTE: A notification of the type just described may have security implications
+if the source routes of the conflict are in different namespaces. The
+notification might include confidential information from one of the namespaces,
+that should not be exposed to an administrative user from the namespace of one
+of the other source routes.
+
+Some specific types of objects and config parameters have unique rules for how
+the parent and child are combined. These are described below.
+
+### Hostname Settings
+
+In the current version of the standard,  `hostname` can be specified at both the
+listener and route level. When both are specified, each is treated as a partial
+FQDN and the two strings are combined at the point of overlap. In this proposal,
+that model is extended to allow `hostname` to be specified at the listener,
+parent route **and** child route levels. Specifying `hostname` at any
+combination of these levels is allowed but the `hostnames` are always combined
+in the sequence: listener, parent route, child route.  
+
+### RouteMatches
+
+The match criteria from the associated sectionName of the parent route’s
+`RouteRule`, are added (logical AND) to each `RouteMatch` in the child route’s
+`RouteRules`. Arrays of matching criteria are evaluated, based on the existing
+rules for that type of array, before being AND'ed with the matching array from
+the other route.
+
+There are specific rules for how some types of matches are combined. These are
+described below.
+
+#### Path Matching
+
+The model described above is simple to understand and apply with all types of
+matching criteria except for path matching. There are two reasons path matching
+is different.
+
+First, the user may want to match routes in the child using paths that are
+relative to the path in the parent route’s match criteria.
+
+The second reason is that there are multiple methods supported for doing path
+matching. These include prefix, postfix and RegEx. Trying to reconcile two path
+matching rules could be very complex and prone to error.
+
+To avoid these issues, the only explicitly supported pattern for using path
+matching, in both a parent and a child route, is using the matching type
+`PathPrefix` in both routes. The two paths are concatenated to form the complete
+path to be matched. When being concatenated. the path from the child is added to
+the end of the parent’s path.
+
+### RouteFilters
+
+The filters specified in the associated sectionName of the parent route’s
+`RouteRules`, are added (logical AND) to each `RouteFilter` in the child route’s
+`RouteRules`.
+
+Configuration conflicts are determined as described in the [Configuration
```

Comment:
What is a conflict? Is it illegal to try to add a header that is already added? To add a header in parent and remove in child? To set the same header in both? To define two different mirror policies in parent/child?

Reply:
This is a good question, maybe we should start with one of the following limitations:

- Filters can not be specified on child routes
- Filters can not be specified on parent routes
- Filters of the same type can only be specified in child or parent route but never both